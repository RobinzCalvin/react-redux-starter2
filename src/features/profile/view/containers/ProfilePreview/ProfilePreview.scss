/* doiuse-disable word-break */
@import '~shared/styles/variables';

.profile-preview {
  &__avatar {
    border-radius: 50%;
    cursor: pointer;
  }

  &__info {
    margin-top: 0.2rem;
    padding: 1rem 1.3rem;
    width: 23rem;
    border-radius: 5px;
    background: #f2f2f2;
    font-size: 0.8rem;

    @media (max-width: $breakpoint-sm) {
      width: 19rem;
    }
  }

  &__main-info {
    display: flex;
    justify-content: space-between;
  }

  &__name {
    word-break: break-all;
    font-size: 1.2rem;
  }

  &__nickname-age {
    margin-left: 1rem;
    width: 5rem;
    text-align: end;
  }

  &__nickname {
    font-weight: 600;
  }

  &__bio {
    overflow: hidden;
    margin: 1rem 0;
    max-height: 6rem;
    text-align: justify;
    text-overflow: ellipsis;

    @media (max-width: $breakpoint-sm) {
      max-height: 3rem;
    }
  }

  &__nickname-age,
  &__bio {
    line-height: 1rem;
  }

  &__edit {
    text-decoration: underline;
    font-size: 0.9rem;
    cursor: pointer;
  }
}
