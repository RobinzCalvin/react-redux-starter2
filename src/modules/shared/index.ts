export { default as Layout } from './Layout/Layout';
export { default as withLayout } from './withLayout';
