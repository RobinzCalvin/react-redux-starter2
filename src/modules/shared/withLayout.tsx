import React from 'react';

import Layout from './Layout/Layout';

function withLayout(title: string) {
  return (Component: React.ComponentType) => () => (
    <Layout title={title}>
      <Component />
    </Layout>
  );
}

export default withLayout;
