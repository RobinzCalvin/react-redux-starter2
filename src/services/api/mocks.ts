import { ICakePreview } from 'shared/types/models';

// tslint:disable:max-line-length
export const cakesPreview: ICakePreview[] =  [
  {
    name: 'Vanilla Almond',
    description: 'Light and sweet but moist and rich in flavor, our Vanilla Almond cake is filled and frosted with our house vanilla buttercream. Vanilla Almond Amycakes are decorated with sparkly clear crystal sprinkles, pearl sprinkles and a ring of buttercream rosettes',
    imageURL: '',
  },
  {
    name: 'Golden Vanilla',
    description: 'Old fashioned, buttery golden yellow cake, just as moist and dense as Grandma used to make. Filled and frosted with our house vanilla buttercream. Golden Vanilla Amycakes are decorated with ascending gold sprinkles, white ganache drizzle and white chocolate curls and flakes',
    imageURL: '',
  },
  {
    name: 'Golden Chocolate',
    description: 'Old fashioned, buttery golden yellow cake, just as moist and dense as Grandma used to make. Filled and frosted with our house chocolate buttercream. Golden Chocolate Amycakes are decorated with ascending gold sprinkles, chocolate ganache drizzle and dark chocolate curls and flakes.',
    imageURL: '',
  },
  {
    name: 'Confetti',
    description: 'Moist Vanilla Almond cake studded with confetti sprinkles in the batter. Confetti Amycakes are frosted with vanilla buttercream and decorated with colorful sprinkles.',
    imageURL: '',
  },
  {
    name: 'Zebra',
    description: 'Alternating layers of vanilla almond and chocolate cakes, filled and frosted with our house vanilla buttercream.  Zebra Amycakes are decorated with chocolate ganache drizzle and white and chocolate curls. ',
    imageURL: '',
  },
];
