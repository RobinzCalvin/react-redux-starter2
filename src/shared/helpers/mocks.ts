import { IProfile, IGithubUser, IDetailedGithubUser, IRepository } from 'shared/types/models';

export function createContainerMock() {
  return function ContainerMock() {
    return null;
  } as any;
}

export function getProfileMock(): IProfile {
  return {
    age: 100,
    nickname: 'Nickname',
    name: 'Name name',
    bio: 'profile bio',
    avatarURL: 'https://the-url.com',
  };
}

export function getGithubUserMock(): IGithubUser {
  return {
    id: 1,
    username: 'username',
    avatarURL: 'http://the-url.com',
    htmlURL: 'http://the-url.com',
  };
}

export function getDetailedGithubUserMock(): IDetailedGithubUser {
  return {
    ...getGithubUserMock(),
    followersNumber: 12,
    followingNumber: 12,
    reposNumber: 12,
    realName: 'The real name',
    location: 'Nowhere',
  };
}

export function getRepositoryMock(): IRepository {
  return {
    id: 1,
    openIssuesNumber: 1,
    starsNumber: 1,
    forksNumber: 1,
    description: 'description',
    name: 'name',
    htmlURL: 'http://the-url.com',
    language: 'en',
    updatedAt: '05/05/2005',
    owner: getGithubUserMock(),
  };
}
// TODO: моки?
