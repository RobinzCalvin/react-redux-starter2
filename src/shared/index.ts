const testContext = (require as any).context('./', true, /^.*\/(?!.*test).*\.tsx?$/);
testContext.keys().forEach(testContext);
