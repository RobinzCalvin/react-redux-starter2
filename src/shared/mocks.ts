import { IProfile, IGithubUser, IDetailedGithubUser, IRepository } from 'shared/types/models';

export const Container: any = () => null;

export const profile: IProfile = {
  age: 100,
  nickname: 'Nickname',
  name: 'Name name',
  bio: 'profile bio',
  avatarURL: 'https://the-url.com',
};

export const githubUser: IGithubUser = {
  id: 1,
  username: 'username',
  avatarURL: 'http://the-url.com',
  htmlURL: 'http://the-url.com',
};

export const detailedGithubUser: IDetailedGithubUser = {
  ...githubUser,
  followersNumber: 12,
  followingNumber: 12,
  reposNumber: 12,
  realName: 'The real name',
  location: 'Nowhere',
};

export const repository: IRepository = {
  id: 1,
  openIssuesNumber: 1,
  starsNumber: 1,
  forksNumber: 1,
  description: 'description',
  name: 'name',
  htmlURL: 'http://the-url.com',
  language: 'en',
  updatedAt: '05/05/2005',
  owner: githubUser,
};
