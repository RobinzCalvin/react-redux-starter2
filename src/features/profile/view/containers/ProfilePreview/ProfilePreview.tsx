import React, { useState, useEffect } from 'react';
import block from 'bem-cn';
import { connect } from 'react-redux';

import { IAppReduxState } from 'shared/types/app';
import { IProfile } from 'shared/types/models';

import { ProfileAvatar } from '../../components';
import { selectors } from '../../../redux';
import './ProfilePreview.scss';

interface IOwnProps {
  onEditClick(): void;
}

interface IStateProps {
  profile: IProfile;
}

type IProps = IOwnProps & IStateProps;

const b = block('profile-preview');

function mapState(state: IAppReduxState): IStateProps {
  return {
    profile: selectors.selectProfile(state),
  };
}

function ProfilePreview(props: IProps) {
  const [isOpen, setIsOpen] = useState<boolean>(false);
  useEffect(documentClickEffect);
  const { profile: { avatarURL, name, nickname, age, bio }, onEditClick } = props;
  const blockRef = React.useRef<HTMLDivElement | null>(null);
  return (
    <div className={b()} ref={blockRef}>
      <div className={b('avatar')} onClick={handleAvatarClick}>
        <ProfileAvatar avatarURL={avatarURL} size="small" />
      </div>
      <div className={b('info', { open: isOpen })}>
        <div className={b('main-info')}>
          <div className={b('name')}>
            {name}
          </div>
          <div className={b('nickname-age')}>
            <div className={b('nickname')}>
              {nickname}
            </div>
            <div className={b('age')}>
              {age} y.o.
            </div>
          </div>
        </div>
        <div className={b('bio')}>
          {bio}
        </div>
        <div className={b('edit')} onClick={onEditClick}>
          Edit
        </div>
      </div>
    </div>
  );

  function documentClickEffect() {
    document.addEventListener('click', handleDocumentClick);
    return () => document.addEventListener('click', handleDocumentClick);
  }

  function handleDocumentClick(e: MouseEvent) {
    if (blockRef.current !== null && !blockRef.current.contains(e.target as Node)) {
      setIsOpen(false);
    }
  }

  function handleAvatarClick() {
    setIsOpen(true);
  }

}

export default connect(mapState)(ProfilePreview);
