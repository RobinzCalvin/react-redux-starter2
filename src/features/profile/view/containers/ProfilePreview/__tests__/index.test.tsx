import React from 'react';
import { mount } from 'enzyme';

import { profile } from 'shared/mocks';

import { ProfilePreview, IProfilePreviewProps } from '../ProfilePreview';

const props: IProfilePreviewProps = {
  onEditClick: jest.fn(),
  profile,
};

describe('ProfilePreview component', () => {
  const component = mount(<ProfilePreview {...props}/>);

  it('should not show profile info initially', () => {
    const info = component.find('.profile-preview__info');
    expect(info.length).toBe(0);
  });

  it('should show profile info on avatar click', () => {
    component.find('.profile-preview__avatar').simulate('click');
    const info = component.find('.profile-preview__info');
    expect(info.length).toBeGreaterThan(0);
  });

  it('should call onEditClick on Edit click', () => {
    component.find('.profile-preview__edit').simulate('click');
    expect(props.onEditClick).toHaveBeenCalledTimes(1);
  });
});
