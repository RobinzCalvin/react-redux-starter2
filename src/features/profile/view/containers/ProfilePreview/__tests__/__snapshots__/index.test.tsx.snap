// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePreview component should match snapshot 1`] = `
<div
  className="profile-preview"
>
  <div
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
    onClick={[Function]}
  >
    <ProfileAvatar
      avatarURL="https://the-url.com"
      size="small"
    />
  </div>
  <WithStyles(Popover)
    anchorEl={null}
    anchorOrigin={
      Object {
        "horizontal": "left",
        "vertical": "bottom",
      }
    }
    onClose={[Function]}
    open={true}
  >
    <div
      className={
        Object {
          "has": [Function],
          "is": [Function],
          "mix": [Function],
          "split": [Function],
          "state": [Function],
          "toString": [Function],
        }
      }
    >
      <div
        className={
          Object {
            "has": [Function],
            "is": [Function],
            "mix": [Function],
            "split": [Function],
            "state": [Function],
            "toString": [Function],
          }
        }
      >
        <div
          className={
            Object {
              "has": [Function],
              "is": [Function],
              "mix": [Function],
              "split": [Function],
              "state": [Function],
              "toString": [Function],
            }
          }
        >
          Name name
        </div>
        <div
          className={
            Object {
              "has": [Function],
              "is": [Function],
              "mix": [Function],
              "split": [Function],
              "state": [Function],
              "toString": [Function],
            }
          }
        >
          <div
            className={
              Object {
                "has": [Function],
                "is": [Function],
                "mix": [Function],
                "split": [Function],
                "state": [Function],
                "toString": [Function],
              }
            }
          >
            Nickname
          </div>
          <div
            className={
              Object {
                "has": [Function],
                "is": [Function],
                "mix": [Function],
                "split": [Function],
                "state": [Function],
                "toString": [Function],
              }
            }
          >
            100
             y.o.
          </div>
        </div>
      </div>
      <div
        className={
          Object {
            "has": [Function],
            "is": [Function],
            "mix": [Function],
            "split": [Function],
            "state": [Function],
            "toString": [Function],
          }
        }
      >
        profile bio
      </div>
      <div
        className={
          Object {
            "has": [Function],
            "is": [Function],
            "mix": [Function],
            "split": [Function],
            "state": [Function],
            "toString": [Function],
          }
        }
        onClick={[MockFunction]}
      >
        Edit
      </div>
    </div>
  </WithStyles(Popover)>
</div>
`;
